File: app/src/main/java/org/xdty/callerinfo/di/modules/MainModule.java
package org.xdty.callerinfo.di.modules;

import org.xdty.callerinfo.contract.MainContract;
import org.xdty.callerinfo.presenter.MainPresenter;

import dagger.Module;
import dagger.Provides;

@Module
public class MainModule {

    private MainContract.View mView;

    public MainModule(MainContract.View view) {
        mView = view;
    }

    @Provides
    MainContract.View provideView() {
        return mView;
    }

    @Provides
    MainContract.Presenter providePresenter() {
        return new MainPresenter(mView);
    }
}

File: app/src/main/java/org/xdty/callerinfo/di/modules/MainModule.kt
package org.xdty.callerinfo.di.modules

import dagger.Module
import dagger.Provides
import org.xdty.callerinfo.contract.MainContract.Presenter
import org.xdty.callerinfo.contract.MainContract.View
import org.xdty.callerinfo.presenter.MainPresenter

@Module
class MainModule(private val mView: View) {
    @Provides
    internal fun provideView(): View {
        return mView
    }

    @Provides
    internal fun providePresenter(): Presenter {
        return MainPresenter(mView)
    }

}